README.md

```diff
@@ -0,0 +1,241 @@
+| kaguya             | (<-- this line should be present)
+| mysql              |
+| performance_schema |
++--------------------+
+4 rows in set (0.00 sec)
+```
+
+Navigate to your installation folder and find the `appsettings.Development.json` file.
+You can do these steps through the command line or through a text editor.
+
+Navigate to the project directory.
+```
+$ cd (installation_directory)/Kaguya/Kaguya/Kaguya
+$ ls
+
+(output)
+appsettings.Development.json  Database/  Global.cs        Migrations/  Program.cs   Startup.cs
+appsettings.json              Discord/   Kaguya.csproj    obj/         Properties/  Web/
+bin/
+```
+
+Copy the contents of the `appsettings.json` file to `appsettings.Development.json`.
+```
+$ cp appsettings.json appsettings.Development.json
+```
+
+Modify `appsettings.Development.json` to include the following:
+- BotToken
+- OwnerId (your discord user ID)
+- Database connection string
+
+
+Open the file in vim, or edit in a text editor (VS Code).
+```bash
+$ vi -v appsettings.Development.json
+or
+$ code appsettings.Development.json
+```
+Your file should now look something like this (note that the "AdminSettings" section will change overtime, this is just an example):
+
+```json
+{
+  "Logging": {
+    "LogLevel": {
+      "Default": "Information",
+      "Microsoft": "Warning",
+      "Microsoft.Hosting.Lifetime": "Information"
+    }
+  },
+  "AllowedHosts": "*",
+  "AdminSettings": {
+    "OwnerId": YOUR_ID, // Optional
+    "OsuApiKey": "" // Optional
+  },
+  "DiscordSettings": {
+    "BotToken": "YOUR_TOKEN",
+    "MessageCacheSize": 50,
+    "AlwaysDownloadUsers": true
+  },
+  "ConnectionStrings": {
+    "Database": "server=localhost;user=root;password=YOUR_PASSWORD;database=kaguya"
+  }
+}
+```
+
+- Save this file and close it.
+
+Now, we need to use entity framework to setup the database for us.
+
+Navigate to the folder containing `Kaguya.csproj` if you have not already done so. Note that this is the same folder as the previous step.
+
+Run:
+```
+$ dotnet ef database update
+
+(output)
+Build started...
+Build succeeded.
+Done.
+```
+
+You are now able to run Kaguya.
+```
+$ dotnet run
+```
+
+## Contributing
+If you wish to contribute to the codebase, please note the following:
+- Create a fork of the code.
+- All commands must adhere to the guidelines below.
+- All changes must be made to the `v4-open-beta` branch. This is the latest stable release.
+- Submit your changes to us through a pull request. In your PR, please give a brief overview of what you changed. Ensure your commits have concise description messages.
+
+### Creating Commands
+To create a Kaguya command, a very specific structure is used. This is so that the `$help` command can properly generate documentation for all commands at runtime.
+A command can be as simple or as complex as necessary, this guide will include both examples.
+
+**Jetbrains Rider C# file template** (all Kaguya Commands are built out from this template):
```

Comment:
@BitMasher I am unsure whether these code templateswork outsie of Rider.

Reply:
i don't know either, probably shouldn't rely on IDE templates

Reply:
Okay. I included it just to showcase what code is required for each new command. Would you say it's beneficial to leave that info in the readme?

Reply:
i would usually make a sample implementation but ya i think it's a good idea to be in the documentation, at a certain point it makes sense to break the readme up into multiple files tho so it's not one giant page

Reply:
Should we make a wiki for this then?